lua/auto-session/init.lua

```diff
@@ -551,11 +582,17 @@ function AutoSession.SaveSessionToDir(session_dir, session_name, show_message)
   session_dir = Lib.validate_root_dir(session_dir)
   Lib.logger.debug("SaveSessionToDir validated session_dir: ", session_dir)
 
+  if not session_name or session_name == "" then
+    -- If no session name is passed in, retrieve the current session name
+    session_name = get_session_name()
```

Comment:
this is the change that causes the side effect that makes manually named sessions "sticky" when calling `SaveSession()`. if no session name is passed, but the `manually_named_session` flag is still true, `get_session_name` will return the last manually saved session name instead of a fresh cwd-driven session name like it does before this change